quinn-proto/src/tests/mod.rs

```diff
@@ -2184,6 +2211,388 @@ fn packet_splitting_not_necessary_after_higher_mtu_discovered() {
     assert_eq!(pair.server.inbound.len(), 1);
 }
 
+#[test]
+fn single_ack_eliciting_packet_triggers_ack_after_delay() {
+    let _guard = subscribe();
+    let mut pair = Pair::default();
+    let (client_ch, _) = pair.connect();
+    pair.drive();
+
+    let stats_after_connect = pair.client_conn_mut(client_ch).stats();
+
+    let start = pair.time;
+    pair.client_conn_mut(client_ch).ping();
+    pair.drive_client(); // Send ping
+    pair.drive_server(); // Process ping
+    pair.drive_client(); // Give the client a chance to process an ack, so our assertion can fail
+
+    // Sanity check: the time hasn't advanced in the meantime)
+    assert_eq!(pair.time, start);
+
+    let stats_after_ping = pair.client_conn_mut(client_ch).stats();
+    assert_eq!(
+        stats_after_ping.frame_tx.ping - stats_after_connect.frame_tx.ping,
+        1
+    );
+    assert_eq!(
+        stats_after_ping.frame_rx.acks - stats_after_connect.frame_rx.acks,
+        0
+    );
+
+    pair.drive();
+    let stats_after_drive = pair.client_conn_mut(client_ch).stats();
+    assert_eq!(
+        stats_after_drive.frame_rx.acks - stats_after_ping.frame_rx.acks,
+        1
+    );
+
+    // The time is start + max_ack_delay
+    assert_eq!(pair.time, start + Duration::from_millis(25));
```

Comment:
Can we avoid an implicit dependency on the default max ack delay value here?

Reply:
Hmm... The ideal thing here would be to configure the endpoint's max ack delay and assert against the provided value. However, it is currently not possible to specify a custom max ack delay, so the default is always used. To keep things simple and explicit, I'll assert against `TransportParameters::default().ack_delay`.

Reply:
nit: Would be nice to have the change in this commit rather than the later one. NBD if the rebase is too much effort though.